### PR TITLE
tests: create partition table in fakeinstaller

### PR DESCRIPTION
**Pull request by mvo5**
Snapd will currently not deal with disks that have no partition table and error. To workaround this limitation the installer will now create the partition table.

### CODE REVIEW

File: tests/lib/fakeinstaller/main.go [hunk 1]
```diff
@@ -45,12 +46,57 @@ func firstVol(volumes map[string]*gadget.Volume) *gadget.Volume {
...
+}
+
 func createPartitions(bootDevice string, volumes map[string]*gadget.Volume) ([]gadget.OnDiskStructure, error) {
 	// TODO: support multiple volumes, see gadget/install/install.go
 	if len(volumes) != 1 {
 		return nil, fmt.Errorf("got unexpected number of volumes %v", len(volumes))
 	}
 
+	vol := firstVol(volumes)
+	// XXX: snapd does not create partition tables so we do it here
```
Comment (alfonsosanchezbeato) [inline tests/lib/fakeinstaller/main.go:94]: Why the XXX? Afaiu snapd will not implement this, or am I missing something?

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks! Just a minor comment.

**Comment by mvo5** [comment]
I tested this locally:
```
$ autopkgtest -sU snapd_2.57.4.dsc -- qemu ~/VM/ubuntu-22.10-64.img 
autopkgtest [19:18:19]: starting date: 2022-10-11
autopkgtest [19:18:19]: version 5.20
autopkgtest [19:18:19]: host top; command line: /usr/bin/autopkgtest -sU snapd_2.57.4.dsc -- qemu /home/egon/VM/ubuntu-22.10-64.img
autopkgtest [19:18:37]: @@@@@@@@@@@@@@@@@@@@ test bed setup
...
2022-10-11 19:40:32 Executing autopkgtest:adt-local:tests/smoke/sandbox (autopkgtest:adt-local) (1/4)...
...
2022-10-11 19:41:27 Discarding autopkgtest:adt-local...
2022-10-11 19:41:27 Successful tasks: 4
2022-10-11 19:41:27 Aborted tasks: 0
+ journalctl --sync
+ journalctl -ab
autopkgtest [19:41:27]: test integrationtests: -----------------------]
autopkgtest [19:41:28]: test integrationtests:  - - - - - - - - - - results - - - - - - - - - -
integrationtests     PASS
autopkgtest [19:41:29]: @@@@@@@@@@@@@@@@@@@@ summary
integrationtests     PASS
```
and it's fine.

File: tests/lib/fakeinstaller/main.go [hunk 2]
```diff
@@ -45,12 +46,57 @@ func firstVol(volumes map[string]*gadget.Volume) *gadget.Volume {
...
+	output, err := exec.Command("blkid", "--probe", "--match-types", "gpt", bootDevice).CombinedOutput()
+	exitCode, err := osutil.ExitCode(err)
+	if err != nil {
+		return err
+	}
+	switch exitCode {
+	case 0:
+		// partition table already exists, nothing to do
+	case 2:
+		// no match found, create partition table
```
Comment (mardy) [inline tests/lib/fakeinstaller/main.go:69]: Should we also use `logger.Noticef` here?

**Pull request merged**